### DISCUSSION TITLE
Reasons for 'DependencyClient' macro not working

### DISCUSSION THREAD

**Discussion by iharandreyev** [pointfreeco/swift-composable-architecture#3480] [Q&A] [answered]
Hello everyone.

I'm stuck in a situation where `DependencyClient` macro does not work.
It does not generate any code, and `Expand Macro` menu item does nothing.

When I create a new file, and paste the following code into it:

```
import ComposableArchitecture

@DependencyClient
struct SomeDependency {
  let doSomething: () -> Int
}

@ObservableState
struct SomeState {
  let value: Int
}
```

`ObservableState` macro does work, and it expands as expected. 
But `DependencyClient` does not.
As soon a I click `Expand Macro`, I hear a click sound, observe no UI changes in Xcode, and when I try again, it's just stuck on `Hide Macro Expansion`

<img width="457" alt="Screenshot 2024-11-05 at 19 23 50" src="https://github.com/user-attachments/assets/cc707743-72b0-400a-9506-35c57109c9ff">

https://github.com/user-attachments/assets/72a45d22-fc29-4c8e-907c-1d1b8e5065c9

I have a feeling that I've messed up my Xcode installation.

Any ideas?

**Comment by mbrandonw** [comment]
Hi @iharandreyev, just because the macro isn't expanding it doesn't mean the macro isn't working. Xcode may be having problems displaying the expansion, but I'm certain that the macro is still working. 

Do you have evidence that shows that the macro is definitely not working? Like a compilation error somewhere? Any code you can share?

**Reply by iharandreyev** [reply]
@mbrandonw no parameter init for `RemoteValueRepo` in the video above is not synthesized by the macro. 
`let a = Self()` does not compile
That's the reason I went asking for help

**Reply by mbrandonw** [reply]
And you have enabled the macro in Xcode? Are you able to provide a minimal project that demonstrates the problem you are seeing?

**Reply by iharandreyev** [reply]
Yes, I did enable the macro. 
Otherwise the project would not compile at all. Now it compiles if I remove no parameter init attempt.

Here's a sample project.
[RemoteValueApp.zip](https://github.com/user-attachments/files/17635114/RemoteValueApp.zip)
One would be interested in `FailingMacro.swift`

**Reply by mbrandonw** [reply, marked as answer]
Thanks for that. The issue is the `let`. If you change it to a `var` it should work.

Currently `let`s are omitted from the initializer because they are given a default value (one that throws the `Unimplemented` error), and so Swift would not be able to set that variable again. We do have some changes planned for the macro that would make this possible, but it's just not how it is today.
